posts/tests.py

```diff
@@ -155,3 +155,44 @@ def test_login_wrong_password(self):
     def test_unregistered_user_login(self):
         response = self.client.post(reverse('login'), self.login3, follow=True)
         self.assertFalse(response.context['user'].is_active)
+
+class LoggedInUserArticleCreationTest(TestCase):
+
+    def setUp(self):
+        self.signup1 = {'username': 'testuser2', 'password1': 'aaaa1234', 'password2': 'aaaa1234'}
+        self.login = {'username': 'testuser2', 'password': 'aaaa1234'}
+        User.objects.create_user(**self.login)
+        self.input = {'title':'What a super day', 'body':'this would have been a great day if...'}
+
+    def test_login_post_creation(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
+
+    def test_User_mine_link_show_post_titles(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
+        response = self.client.get(reverse('posts:myposts'), follow=True)
+        title = 'What a super day'
+        body = 'this would have been a great day if...'
+        self.assertContains(response, title, status_code=200)
+        self.assertNotContains(response, body, status_code=200)
+
+    def test_anonymous_user_has_no_mine_link(self):
+        response = self.client.post(reverse('login'), self.login, follow=True)
+        self.assertTrue(response.context['user'].is_active)
+        response = self.client.post(reverse('posts:create'), self.input, follow=True)
+        instance = Post.objects.get()
+        self.assertRedirects(response, reverse('posts:post', kwargs={'id': instance.id}), status_code=302,
+                             target_status_code=200, fetch_redirect_response=True)
```

Comment:
what is the use of lines 189 - 194 in this function

Reply:
it was unnecessary , i thought of creating a post and then loggin oout and checking for '/mine' link
i have removed that in the coming pr